hs_core/views/resource_folder_hierarchy.py

```diff
@@ -452,3 +457,199 @@ def data_store_file_or_folder_move_or_rename(request, res_id=None):
+    tgt_path = resolve_request(request).get('target_path', None)
+    if src_path is None or tgt_path is None:
+        return HttpResponse('Bad request - src_path or tgt_path is not included',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not src_path or not tgt_path:
+        return HttpResponse('Bad request - src_path or tgt_path cannot be empty',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    src_path = str(src_path).strip()
+    tgt_path = str(tgt_path).strip()
+    src_folder, src_base = os.path.split(src_path)
+    tgt_folder, tgt_base = os.path.split(tgt_path)
+    if src_folder != tgt_folder:
+        return HttpResponse('Bad request -- src_path and tgt_path must be in same folder',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not src_path.startswith('data/contents/'):
+        return HttpResponse('Bad request - src_path must start with data/contents/',
+                            status=status.HTTP_400_BAD_REQUEST)
+    if src_path.find('/../') >= 0 or src_path.endswith('/..'):
+        return HttpResponse('Bad request - src_path cannot contain /../',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not tgt_path.startswith('data/contents/'):
+        return HttpResponse('Bad request - tgt_path must start with data/contents/',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if tgt_path.find('/../') >= 0 or tgt_path.endswith('/..'):
+        return HttpResponse('Bad request - tgt_path cannot contain /../',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    istorage = resource.get_irods_storage()
+
+    # check that the target doesn't exist
+    tgt_folder, tgt_base = os.path.split(tgt_path)
+    tgt_folder = tgt_folder[len('data/contents/'):]
+    tgt_abspath = get_path(resource, tgt_base, folder=tgt_folder)
+    if istorage.exists(tgt_abspath):
+        return HttpResponse('Bad request - tgt_path already exists', 
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    # check that the source exists
+    src_folder, src_base = os.path.split(src_path)
+    src_folder = src_folder[len('data/contents/'):]
+    src_abspath = get_path(resource, src_base, folder=src_folder)
+    if istorage.exists(src_abspath):
+        return HttpResponse('Bad request - src_path doesn't exist', 
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        rename_file_or_folder(user, res_id, src_path, tgt_path)
+    except SessionException as ex:
+        return HttpResponse(ex.stderr, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except DRF_ValidationError as ex:
+        return HttpResponse(ex.detail, status=status.HTTP_400_BAD_REQUEST)
+
+    return_object = {'target_rel_path': tgt_path}
+
+    return HttpResponse(
+        json.dumps(return_object),
+        content_type='application/json'
+    )

```

Comment:
@alvacouch Agreed we need to have a backend function to support move for a list of source files to enforce UI and REST atomicity, and agreed this is a valid case to split move and rename since rename only needs to support one file rename at a time, and move needs to support multiple file move at a time. The same istorage.moveFile() base method can be used for both move and rename, and we can build move and rename methods to support our use cases.